### PR TITLE
[Merged by Bors] - feat(FieldTheory/IntermediateField): add `Subfield.extendScalars`

### PR DESCRIPTION
**Pull request by acmepjz**
... which can be viewed as an inverse to `IntermediateField.toSubfield`.

This is similar to `IntermediateField.extendScalars`.

 
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/FieldTheory/IntermediateField.lean [hunk 1]
```diff
@@ -605,11 +605,63 @@ example {F : IntermediateField K L} {E : IntermediateField F L} : Algebra K E :=
...
 
+section Subfield
+
+open Subfield
+
+variable {F E E' : Subfield L} (h : F ≤ E) (h' : F ≤ E') {x : L}
+
+/-- If `F ≤ E` are two subfields of `L`, then `E` is also an intermediate field of
+`L / F`. It can be viewed as an inverse to `IntermediateField.toSubfield`. -/
+def _root_.Subfield.extendScalars : IntermediateField F L :=
```
Comment (alreadydone) [inline Mathlib/FieldTheory/IntermediateField.lean:616]: Can you close the outer namespace and use `namespace Subfield` for everything?

Comment (acmepjz) [reply Mathlib/FieldTheory/IntermediateField.lean]: Done.

File: Mathlib/FieldTheory/IntermediateField.lean [hunk 2]
```diff
@@ -603,13 +603,67 @@ end RestrictScalars
...
+
+theorem extendScalars_le_iff (E' : IntermediateField F L) :
+    extendScalars h ≤ E' ↔ E ≤ E'.toSubfield := Iff.rfl
+
+theorem le_extendScalars_iff (E' : IntermediateField F L) :
+    E' ≤ extendScalars h ↔ E'.toSubfield ≤ E := Iff.rfl
+
+variable (F)
+
+/-- `Subfield.extendScalars` is an order isomorphism from
```
Comment (jcommelin) [inline Mathlib/FieldTheory/IntermediateField.lean:639]:
```suggestion
/-- `Subfield.extendScalars.orderIso` bundles `Subfield.extendScalars`
into an order isomorphism from
```

**Review by jcommelin — COMMENTED** [review]
bors d+

**Pull request closed without merging**